subsystem/host_test.go

```diff
@@ -593,6 +594,11 @@ var _ = Describe("Host tests", func() {
 		})
 		Expect(err).NotTo(HaveOccurred())
 
+		// confirm if new registration updated the timestamp
+		h2 = getHostV2(*infraEnvID2, *hostID)
+		h2newRegistrationTimestamp := h2.RegisteredAt
+		Expect(h2newRegistrationTimestamp).NotTo(Equal(h2initialRegistrationTimestamp))
```

Comment:
I'd be careful of time comparisons generally.

Maybe `Expect(h2newRegistrationTimestamp.Equal(h2initialRegistrationTimestamp)).To(BeFalse())` would be a bit more reliable.

Reply:
Just because you were seeing a test failure.

Reminded me of https://github.com/openshift/assisted-service/pull/3041 which is slightly different (because strings) but still worth thinking about.

Reply:
I think that's not the reason... I realized that there was a hidden internal function that was handling the scenario when a host with specified ID already exists in the DB. If that's the case, it was transparently ignoring everything apart from `host.Kind`. Now added that it also handles `host.RegisteredAt`, let's see the test